crates/notion-core/src/tool/mod.rs

```diff
@@ -81,23 +79,14 @@ impl Display for ToolSpec {
     }
 }
 
-#[derive(Debug, Fail, NotionFail)]
-#[fail(display = "{}", error)]
-#[notion_fail(code = "ExecutionFailure")]
-pub(crate) struct BinaryExecError {
-    pub(crate) error: String,
-}
-
-impl BinaryExecError {
-    pub(crate) fn from_io_error(error: &io::Error) -> Self {
-        if let Some(inner_err) = error.get_ref() {
-            BinaryExecError {
-                error: inner_err.to_string(),
-            }
-        } else {
-            BinaryExecError {
-                error: error.to_string(),
-            }
+fn from_io_error(error: &io::Error) -> ErrorDetails {
```

Comment:
Similar to symlink_error, this should be named something like `binary_exec_error`.